src/map/entities/trustentity.cpp

```diff
@@ -59,11 +60,26 @@ void CTrustEntity::PostTick()
     if (loc.zone && updatemask && status != STATUS_DISAPPEAR)
     {
         loc.zone->PushPacket(this, CHAR_INRANGE, new CEntityUpdatePacket(this, ENTITY_UPDATE, updatemask));
+        for (auto PTrust : ((CCharEntity*)PMaster)->PTrusts)
+        {
+            if (PTrust == this)
+            {
+                ((CCharEntity*)PMaster)->pushPacket(new CTrustSyncPacket((CCharEntity*)PMaster, this));
+            }
+        }
 
-        if (PMaster && PMaster->PPet == this)
+        if (updatemask & UPDATE_HP)
         {
-            ShowDebug("CTrustSyncPacket");
-            ((CCharEntity*)PMaster)->pushPacket(new CTrustSyncPacket((CCharEntity*)PMaster, this));
+            if (PMaster->PParty != nullptr)
+            {
+                for (auto PEntity : ((CCharEntity*)PMaster)->PParty->members)
```

Comment:
Ideally I'd use `ForAlliance([&](auto PEntity)` like the char form of this, but I couldn't get it to recognise any party members...

Reply:
not sure why that wouldnt work since what you have atm is the equivalent of ForParty
https://github.com/DarkstarProject/darkstar/blob/master/src/map/entities/battleentity.h#L545
```cpp
PMaster->ForParty([this](auto PMember)
{
    if (PMember->objtype == TYPE_PC)
        static_cast<CCharEntity*>(PEntity)->pushPacket(new CCharHealthPacket(this));
});
```